src/GraphQL/Language/AST/Fields.cs

```diff
@@ -24,31 +28,22 @@ private sealed class DefaultFieldCollectionStrategy : IFieldCollectionStrategy
         /// <br/><br/>
         /// See http://spec.graphql.org/June2018/#sec-Field-Collection and http://spec.graphql.org/June2018/#CollectFields()
         /// </summary>
-        /// <param name="context"></param>
-        /// <param name="specificType"></param>
-        /// <param name="selectionSet"></param>
-        /// <param name="fieldCollectionStrategy">
-        /// If set to <see langword="null"/>, <see cref="ExecutionHelper.ShouldIncludeNode"/> will be used to work
-        /// as required by the specification. Set this parameter if you understand exactly what you are doing,
-        /// because your actions may lead to the fact that the server's behavior ceases to comply with the
-        /// specification requirements.
-        /// </param>
-        public Fields CollectFrom(ExecutionContext context, IGraphType specificType, SelectionSet selectionSet, IFieldCollectionStrategy fieldCollectionStrategy = null)
+        public Fields CollectFrom(ExecutionContext context, IGraphType specificType, SelectionSet selectionSet)
         {
             List<string> visitedFragmentNames = null;
-            CollectFields(context, specificType, selectionSet, fieldCollectionStrategy ?? DefaultFieldCollectionStrategy.Instance, ref visitedFragmentNames);
+            CollectFields(context, specificType, selectionSet, context.ExecutionStrategy ?? DefaultExecutionStrategy.Instance, ref visitedFragmentNames);
```

Comment:
Do we need to make context.ExecutionStrategy optional? I’d remove the “DefaultExecutionStrategy” altogether. If nothing else just make it pull the ParallelExecutionStrategy instance.

Reply:
Done